model/minsky.cc

```diff
@@ -713,6 +726,96 @@ namespace minsky
        });
     canvas.requestRedraw();
   }
+  
+  /*
+  For using GSL non-linear least square fitting routines
+  patterned on first Large Nonlinear Least Squares Example.
+  See online documentation at https://www.gnu.org/software/gsl/doc/html/nls.html
+  */
+      
+namespace
+{	
+
+   int residualFunc(const gsl_vector * x, void *params,
+          gsl_vector * f)
+   {
+	if (params==NULL) return GSL_EBADFUNC;
+	try { 	  
+		
+	      auto pars=static_cast<OptParams*>(params);  
+	      
+          int err = GSL_SUCCESS;
+          auto n = ValueVector::stockVars.size();
+          vector<double> xx(n);
+          
+          memcpy(xx.data(),reinterpret_cast<const vector<double>*>(x),n);
```

Comment:
This is wrong. A gsl_vector is most categorically not a vector of double. It might accidentally work if the address of the gsl_vector's data coincides with its address, but that doesn't strike me as very likely. How would it know how long it is, fo instance?

Reply:
Okay, I don't know why I thought I had to recast the gsl_vector x to the same type as xx, which is of type vector<double>, because it "works" as just memcpy(xx.data(),x,n). But, then again, it doesn't "work" when I use memcpy(xx.data(),x,sizeof(x)) as I'm supposed to? So, this makes me doubt the implementation now...